plugins/PrivacyManager/API.php

```diff
@@ -234,29 +228,114 @@ private function formatAvailableColumnsToAnonymize($columns)
+                $file = StaticContainer::get('Piwik\Plugins\CustomJsTracker\TrackerUpdater')->getToFile();
+                $filename = $file->getName();
+            }
+
+            return [$filename, $file->hasWriteAccess()];
+        }
+
+        return ['', false];
+    }
+
+    /**
+     * Provide anonymisation settings to Matomo UI only
+     *
      * @internal
      */
-    public function setAnonymizeIpSettings($anonymizeIPEnable, $maskLength, $useAnonymizedIpForVisitEnrichment, $anonymizeUserId = false, $anonymizeOrderId = false, $anonymizeReferrer = '', $forceCookielessTracking = false, $randomizeConfigId = false)
+    public function getAnonymisationSettings(?int $idSiteSpecific = null): array
     {
+        if (is_numeric($idSiteSpecific)) {
+            $idSite = intval($idSiteSpecific);
+            Piwik::checkUserHasAdminAccess($idSiteSpecific);
+        } else {
+            $idSite = null;
+            Piwik::checkUserHasSuperUserAccess();
+        }
+
+        $privacyConfig = new Config($idSite);
+        $settings = [];
+        foreach ($privacyConfig->getConfigPropertyNames() as $propertyName) {
+            $settings[$propertyName] = $privacyConfig->{$propertyName};
+        }
+        $settings['useSiteSpecificSettings'] = $privacyConfig->useSiteSpecificSettings();
+
+        // provide extra settings
+        [$trackerFilename, $trackerFileWritable] = $this->getTrackerFileDetails();
+        $settings = array_merge($settings, [
+            'maskLengthOptions' => PrivacyManager::getMaskLengthOptions(),
+            'useAnonymizedIpForVisitEnrichmentOptions' =>
+                PrivacyManager::getUseAnonymizedIpForVisitEnrichmentOptions(),
+            'referrerAnonymizationOptions' => ReferrerAnonymizer::getAvailableAnonymizationOptions(),
+            'configRandomisationFeatureFlag' =>
+                $this->featureFlagManager->isFeatureActive(ConfigIdRandomisation::class),
+            'trackerFileName' => $trackerFilename,
+            'trackerWritable' => $trackerFileWritable,
+        ]);
+
+        return $settings;
+    }
+
+    /**
+     * @internal
+     */
+    public function setAnonymisationSettings(
+        $enableIpAnonymizer,
+        $ipAddressMaskLength,
+        $useAnonymizedIpForVisitEnrichment,
+        $anonymizeUserId = false,
+        $anonymizeOrderId = false,
+        $anonymizeReferrer = '',
+        $forceCookielessTracking = false,
+        $randomizeConfigId = false,
+        $idSiteSpecific = null,
+        $useSiteSpecificSettings = false
+    ) {
         Piwik::checkUserHasSuperUserAccess();
 
-        if ($anonymizeIPEnable == '1') {
-            IPAnonymizer::activate();
-        } elseif ($anonymizeIPEnable == '0') {
-            IPAnonymizer::deactivate();
+        if (is_numeric($idSiteSpecific)) {
+            $idSite = intval($idSiteSpecific);
+            Piwik::checkUserHasAdminAccess($idSiteSpecific);
```

Comment:
There is still a  `checkUserHasSuperUserAccess` call right before 👀

Reply:
Moved that to inside the condition so only one check is performed based on if it's global or site-specific. Admin check will be true for all super admins early on.